apps/charterafrica/src/pages/api/v1/resources/ecosystem/[action].page.js

```diff
@@ -0,0 +1,43 @@
+import { airtableSchema } from "@/charterafrica/lib/ecosystem/airtable";
+
+const isApiKeyValid = (key) => {
+  return key && key === process.env.RESOURCES_SECRET_TOKEN;
+};
+
+export async function schema(req, res) {
+  const { source } = req.query;
+  const sourceMap = { airtable: airtableSchema };
+  const schemaFunc = sourceMap[source];
+  if (schemaFunc) {
+    const response = await schemaFunc(req, res);
+    if (response) {
+      return res.status(200).json(response);
+    }
+  }
+  return res.status(404).json({ message: "SOURCE_NOT_FOUND" });
+}
+
+const actionMap = {};
+
+export default async function handler(req, res) {
+  const {
+    query: { action },
+  } = req;
+  // Because we do not need to expose API KEY
+  if (action === "schema") {
+    return schema(req, res);
+  }
```

Comment:
Use the `actionMap` above.

Reply:
Are we going to share SECRET_KEY on payloadCMS admin console?

Reply:
May be I'm missing something... context please.

Reply:
I think this just makes the code a bit harder to follow @koechkevin ... Since every `actionFunc` receives `req`, they can check if valid key exists if they need to protect the `action`.